### PR TITLE
NMS-14498: Parameter to control requisition location

### PR DESCRIPTION
**Pull request by fooker**

### CODE REVIEW

**Review by amay94 — DISMISSED** [review]

File: plugin/src/main/java/org/opennms/velocloud/requisition/CustomerRequisitionProvider.java [hunk 1]
```diff
@@ -82,7 +84,8 @@ protected Requisition handleRequest(final RequestContext context) throws Veloclo
         for (var edge : client.getEdges()) {
             final var node = ImmutableRequisitionNode.newBuilder()
                                                      .setForeignId(edge.name)
-                                                     .setNodeLabel(edge.name);
+                                                     .setNodeLabel(edge.name)
+                                                     .setNodeLabel(context.getLocation());
```
Comment (aramosvizcarra) [inline plugin/src/main/java/org/opennms/velocloud/requisition/CustomerRequisitionProvider.java:88]:
```suggestion
                                                     .setLocation(context.getLocation());
```

File: plugin/src/main/java/org/opennms/velocloud/requisition/PartnerRequisitionProvider.java [hunk 2]
```diff
@@ -78,7 +80,8 @@ protected Requisition handleRequest(final RequestContext context) throws Veloclo
             for (final var gateway : client.getGateways()) {
                 final var node = ImmutableRequisitionNode.newBuilder()
                                                          .setForeignId(gateway.gatewayName)
-                                                         .setNodeLabel(gateway.gatewayName);
+                                                         .setNodeLabel(gateway.gatewayName)
+                                                         .setNodeLabel(context.getLocation());
```
Comment (aramosvizcarra) [inline plugin/src/main/java/org/opennms/velocloud/requisition/PartnerRequisitionProvider.java:84]:
```suggestion
                                                         .setLocation(context.getLocation());
```

**Review by aramosvizcarra — CHANGES_REQUESTED** [review]
To small changes

**Review by aramosvizcarra — COMMENTED** [review]
Shouldn't the license be now 2023?

File: plugin/src/main/java/org/opennms/velocloud/requisition/CustomerRequisitionProvider.java (hunk 1 above)
Comment (fooker) [reply plugin/src/main/java/org/opennms/velocloud/requisition/CustomerRequisitionProvider.java]: Fixed

**Review by amay94 — DISMISSED** [review]

**Review by christianpape — DISMISSED** [review]
LGTM.

File: docs/modules/configuration/pages/requisitions.adoc [hunk 3]
```diff
@@ -5,18 +5,20 @@ In order to monitor the Velocloud gateways, edges, and links, you need to import
 To do so, create a requisition for a given connection alias.
 A connection alias can represent a customer connection or a partner connection.
 While a customer connection only allows you to query for entities associated with the customer itself, a partner connection has access to top-level entities that are shared by customers.
+All nodes created by a requisition are assigned to the same location controlled by the `location' parameter, which defaults to the default location if not specified.
```
Comment (Bonrob2) [inline docs/modules/configuration/pages/requisitions.adoc:8]:
```suggestion
All nodes created by a requisition are assigned to the same location controlled by the `location' parameter, which uses the default location if not specified.
```

**Review by Bonrob2 — COMMENTED** [review]
Just a minor change to minimize the number of times we say "default". :-)

**Review by aramosvizcarra — APPROVED** [review]

**Review by christianpape — DISMISSED** [review]
LGTM.

**Review by Bonrob2 — APPROVED** [review]
Docs LGTM!

**Review by christianpape — DISMISSED** [review]
LGTM.

**Review by christianpape — APPROVED** [review]

**Pull request merged**